doc/ingest-proposal.md

```diff
@@ -0,0 +1,102 @@
+# Proposal for a new Ingest design
+
+## Current Problems
+
+1. Not resilient to process death.
+2. Half the logic is in storage writer the other half in a block hook.
+3. Undefined behavior if multiple publishers sync the same chain.
+4. Relies of a chain of blocking channels to reverse the order of graphsync.
+5. not linear execution flow.
+
+The indexer semantics are relatively simple, yet the ingester may fail in
+surprising and hard to reason about ways. The goal of this proposal is to
+outline a design for an alternate ingester that upholds the indexer semantics
+while being easy to reason about.
+
+## Overview of the network
+
+The indexer ingests data from publishers. There can be one publisher per
+provider or many publishers per provider. Additionally there can be faulty or
+malicious publishers that can point to other chains.
```

Comment:
Does this account for one publisher for many providers?

Reply:
A publisher only publishes a single chain. Are you saying that there may be multiple providers within a single chain?

Reply:
Yes. If by this you mean that one publisher is managing one chain and that chain may have many providers.

A publisher (as it is designed right now) can not have multiple chains. There is a 1:1 mapping of publisher to chain. Of course a publisher may flip between chains but it would only advertise a single chain in its head.